skrifa/src/outline/glyf/mod.rs

```diff
@@ -827,25 +937,267 @@ impl<'a> Scaler<'a> {
+            *contour = end_pt.get();
+        }
+        // Adjust the running point/contour total counts
+        self.point_count += point_count;
+        self.contour_count += contour_count;
+        // Append phantom points to the outline.
+        for (i, phantom) in self.phantom.iter().enumerate() {
+            points[phantom_start + i] = *phantom;
+            flags[phantom_start + i] = Default::default();
+        }
+        // Acquire deltas
+        if self.outlines.gvar.is_some() && !self.coords.is_empty() {
+            let gvar = self.outlines.gvar.as_ref().unwrap();
+            let glyph = deltas::SimpleGlyph {
+                points: &mut points[..],
+                flags: &mut flags[..],
+                contours,
+            };
+            let deltas = self
+                .memory
+                .deltas
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            let iup_buffer = self
+                .memory
+                .iup_buffer
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            if deltas::simple_glyph(
+                gvar,
+                glyph_id,
+                self.coords,
+                self.outlines.has_var_lsb,
+                glyph,
+                iup_buffer,
+                deltas,
+            )
+            .is_ok()
+            {
+                for (point, delta) in points.iter_mut().zip(deltas) {
+                    *point += *delta;
+                }
+            }
+        }
+        // Apply scaling
+        if self.is_scaled {
+            let scale = self.scale.to_f32();
+            for point in points.iter_mut() {
+                *point = point.map(|c| c * scale);
+            }
+        }
+
+        if points_start != 0 {
+            // If we're not the first component, shift our contour end points.
+            for contour_end in contours.iter_mut() {
+                *contour_end += points_start as u16;
+            }
+        }
+        Ok(())
+    }
+
+    fn load_composite(
+        &mut self,
+        glyph: &CompositeGlyph,
+        glyph_id: GlyphId,
+        recurse_depth: usize,
+    ) -> Result<(), DrawError> {
+        use DrawError::InsufficientMemory;
+        let scale = self.scale.to_f32();
+        // The base indices of the points for the current glyph.
+        let point_base = self.point_count;
+        // Compute the per component deltas. Since composites can be nested, we
+        // use a stack and keep track of the base.
+        let mut have_deltas = false;
+        let delta_base = self.component_delta_count;
+        if self.outlines.gvar.is_some() && !self.coords.is_empty() {
+            let gvar = self.outlines.gvar.as_ref().unwrap();
+            let count = glyph.components().count() + PHANTOM_POINT_COUNT;
+            let deltas = self
```

Comment:
HB diff: same as simple-- floating point delta computation

Reply:
Deferred to https://github.com/googlefonts/fontc/issues/784